### PR TITLE
Implemented ios authentication

### PR DESCRIPTION
**Pull request by ziccardi**
## Motivation

Implement XAMARIN OIDC integration for iOS

JIRA: https://issues.jboss.org/browse/AEROGEAR-2461

## Description

Platform dependent implementation of oidc integration for iOS
## Progress

- [x] Implement OIDCAuthenticator
- [x] Implement Option class
- [x] Add comments
- [x] Test

### CODE REVIEW

File: Auth/Auth.Platform.iOS/Authenticator/OIDCAuthenticator.cs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                    authenticateTaskComplete.TrySetResult(User.NewUser().FromUnverifiedCredential(credential, this.keycloakConfig.ResourceId));
+                } else {
+                    authenticateTaskComplete.TrySetException(new Exception(error.LocalizedDescription));
+                }
+            });
+
+            return authenticateTask;
+        }
+
+        public async Task<bool> Logout(User currentUser)
```
Comment (wei-lee) [inline Auth/Auth.Platform.iOS/Authenticator/OIDCAuthenticator.cs]: looks like we can create an abstract base class in the Auth.Core module to share some of the code? WDYT?

Comment (ziccardi) [reply Auth/Auth.Platform.iOS/Authenticator/OIDCAuthenticator.cs]: Totally agree

File: Auth/Auth.Platform.iOS/Authenticator/OIDCAuthenticator.cs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        {
+            nonNull(currentUser, "current user");
+            string identityToken = currentUser.IdentityToken;
+            var logoutUrl = keycloakConfig.LogoutUrl(identityToken, authenticationConfig.RedirectUri.ToString());
+
+            TaskCompletionSource<bool> completionSource = new TaskCompletionSource<bool>();
+            IHttpResponse response = await httpService.NewRequest().Get(logoutUrl).Execute();
```
Comment (wojta) [inline Auth/Auth.Platform.iOS/Authenticator/OIDCAuthenticator.cs]: using `var` may be better in this case I think. It's already called response, so you know what to expect in the code.

Comment (ziccardi) [reply Auth/Auth.Platform.iOS/Authenticator/OIDCAuthenticator.cs]: @wojta I don't think that the returned type from `Execute` is so evident when reading the code. From the point of view of readability, I prefer to leave the explicit `IHttpResponse` declaration.

Comment (wojta) [reply Auth/Auth.Platform.iOS/Authenticator/OIDCAuthenticator.cs]: yeah, maybe not, I wrote the code of `Execute()` so it's obvious to me  :)

**Comment by ziccardi** [comment]
@wei-lee @wojta mind giving another look?

**Pull request merged**